### PR TITLE
7pm d custom api

### LINKED ISSUE #37: Users can add their google search api to their accounts
# User Story

As a user I can add a custom google search api token to my account, so that I can enable searching from within the mapache search app for up to 100 searches per user per day.

# Acceptance Criteria

- [ ] Each user will be able to submit up to 100 searches per day
- [ ] Each user will be able to associate a custom google search api token with their unique account

# Implementation Todos

Front end:
- [x] Implement a field to input a custom google search api token
- [ ] Display error message for invalid token

Back end:
- [ ] Create variable/object to store unique custom google search api tokens with each account
- [ ] Modify MongoDB objects in order to account for extra api token field
- [ ] Store variable/object in MongoDB for each account
- [ ] Implement tests and error message for invalid token

### PR DESCRIPTION
**Pull request by soohyunchoi**
- Added frontend for custom api input fields and user information
- Added tests for frontend rendering
- This pull request **partially** resolves #37 
Screen Shot 2020-12-09 at 11 56 54 PM
- Todo: implement backend and functions to check if user token is valid/null

### CODE REVIEW

File: src/main/java/edu/ucsb/mapache/controllers/RoleController.java [hunk 1]
```diff
@@ -99,4 +100,32 @@
...
+
+<<<<<<< HEAD
+  @PutMapping(value = "/api/addApiKey", produces = "application/json")
+  public ResponseEntity<String> setApiToken(@RequestHeader("Authorization") String authorization, 
+  @RequestBody @Valid String token) 
+=======
+  @PutMapping("/apiKey/{token}")
+  public ResponseEntity<String> setApiToken(@RequestHeader("Authorization") String authorization, 
+  @PathVariable("token") String token) 
+>>>>>>> 6dca1329b74d108ad745acb32b232132ce3bb751
```
Comment (andrewhlu) [inline src/main/java/edu/ucsb/mapache/controllers/RoleController.java:124]: Diff markers present here

File: src/main/java/edu/ucsb/mapache/entities/AppUser.java [hunk 2]
```diff
@@ -19,10 +19,16 @@
   private String firstName;
   @Column(nullable = false)
   private String lastName;
+<<<<<<< HEAD
   @Column(nullable = false, columnDefinition = "integer default 100") 
   private int searchRemain=100;
   @Column(nullable = false)
   private long time;
+=======
+>>>>>>> 6dca1329b74d108ad745acb32b232132ce3bb751
```
Comment (andrewhlu) [inline src/main/java/edu/ucsb/mapache/entities/AppUser.java:28]: Diff markers present here

File: src/main/java/edu/ucsb/mapache/entities/AppUser.java [hunk 3]
```diff
@@ -92,6 +99,62 @@ public String toString() {
...
+    this.firstName = firstName;
+    return this;
+  }
+
+  public AppUser lastName(String lastName) {
+    this.lastName = lastName;
+    return this;
+  }
+
+>>>>>>> 6dca1329b74d108ad745acb32b232132ce3bb751
```
Comment (andrewhlu) [inline src/main/java/edu/ucsb/mapache/entities/AppUser.java:152]: Diff markers present here.

**Review by andrewhlu — CHANGES_REQUESTED** [review]
Looks like you tried to merge / rebase but you didn't resolve all the merge conflicts and instead committed them. Please resolve this and re-request the staff for a review. Please also ensure that your tests pass.

**Comment by soohyunchoi** [comment]
> Looks like you tried to merge / rebase but you didn't resolve all the merge conflicts and instead committed them. Please resolve this and re-request the staff for a review. Please also ensure that your tests pass.

Sorry about that. All frontend tests are passing with 100% code coverage. I can't test the backend but it should be working with all tests passing after I take out these diff markers.

**Comment by andrewhlu** [comment]
Thanks for fixing the merge conflict. As you mentioned, it looks like your test coverage isn't at 100% - could you fix this and then re-request the staff again?

**Comment by pconrad** [comment]
Rolled over to issue #135 in W21

**Pull request closed without merging**